### PR TITLE
Sc-19455: Topic Query Result View 

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This PR adds the Topic query result view component and the default binary result display component 

[Front-End: include link to Figma designs]

Fixes SC-XXXXX

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

[Copy acceptance criteria checklist from story for reviewer, or add a brief acceptance criteria here]

[Front-End: add screenshots/videos of changes made]

image

### Definition of Done

- [x] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [ ] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [x] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  console.log('[] mimetype', mimeType);
+
+  return (
+    <div className="shadow-md h-20 max-h-80 overflow-y-auto bg-black p-4 text-white">
+      <pre className="mx-auto text-sm">
+        {result && <DisplayResultData result={result} mimeType={mimeType} />}
+        {!result && t`No results found`}
+        {!result &&
+          !mimeType &&
+          t`No query result. Try the default query or enter your own query. See EnSQL documentation for example queries.`}
```
Comment (masskoder) [inline web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx:21]: This message is suggested by @eschmier

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const QueryResultContent: React.FC<QueryResultContentProps> = ({ result, mimeType }) => {
+  console.log('[] result', result); // added this avoid eslint error
+  console.log('[] mimetype', mimeType);
+
+  return (
+    <div className="shadow-md h-20 max-h-80 overflow-y-auto bg-black p-4 text-white">
+      <pre className="mx-auto text-sm">
+        {result && <DisplayResultData result={result} mimeType={mimeType} />}
+        {!result && t`No results found`}
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx:18]: I noticed that this message and the one below currently display in the screenshot that was provided. Shouldn't only one of them appear?

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  mimeType: string;
+}
+
+const QueryResultContent: React.FC<QueryResultContentProps> = ({ result, mimeType }) => {
+  console.log('[] result', result); // added this avoid eslint error
+  console.log('[] mimetype', mimeType);
+
+  return (
+    <div className="shadow-md h-20 max-h-80 overflow-y-auto bg-black p-4 text-white">
+      <pre className="mx-auto text-sm">
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx:16]: The text is difficult for me to read on the screen. Could the base font be used instead?

**Review by daniellemaxwell — APPROVED** [review]
The view looks good and it's all coming together nicely!

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  // comment out the above two lines and uncomment the below two lines to test the default result
+  result = result ?? mockResult;
+  mimeType = mimeType ?? mockMimeType;
+
+  return (
+    <div className="shadow-md h-20 max-h-80 overflow-y-auto bg-black p-4 text-white">
+      <pre className="mx-auto text-sm">
+        <code>
+          {result && <DisplayResultData result={result} mimeType={mimeType} />}
+          {!result && t`No results found`}
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx:26]:
```suggestion
          {!result && t`No results found.`}
```

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx (hunk 2 above)
Comment (masskoder) [reply web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx]: Good catch, I improve this part.

File: web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx (hunk 3 above)
Comment (masskoder) [reply web/beacon-app/src/features/topics/components/TopicQueryResult/QueryResultContent.tsx]: Sure we could try the base font for now. with the real data, we could see if it's the best size.

**Pull request merged**